### ISSUE TITLE
[ec2] Add alarms supporting for EC2 instances

### ISSUE THREAD

**Issue by ayush987goyal** [cdklabs/cdk-monitoring-constructs#45]

**Comment by ayush987goyal** [comment]
From @voho 

> Good idea! I just sense a problem in situations we have a search query. We could add the alarms if you either specify the instance IDs manually or have something like an ASG.
  What kind of input or API would you prefer for this?


Potentially dependent on #43

**Comment by vumdao** [comment]
Hi, Any idea how we set alarm for EC2 instance, specifically per instance id or ASG?

**Comment by tmegow** [comment]
Alarms for EC2 instances is a desirable feature for our team. Where does this goal sit on the development roadmap?

Additional question: Can [Cloudwatch Agent metrics](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/metrics-collected-by-CloudWatch-agent.html) be (optionally) included in the EC2MonitoringProps?

**Comment by echeung-amzn** [comment]
> Where does this goal sit on the development roadmap?

Hi, we don't have a concrete roadmap but ad-hoc contributions are welcome!

> Can [Cloudwatch Agent metrics](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/metrics-collected-by-CloudWatch-agent.html) be (optionally) included in the EC2MonitoringProps?

Please open a separate issue for this, ideally with an idea of what a desired API might look like.

**Comment by echeung-amzn** [comment]
Duplicate of #515
